db/rdbms/src/main/resources/mapper/ProcessInstanceMapper.xml

```diff
@@ -69,20 +69,53 @@
     ${paging.after}
   </select>
 
+  <!-- TODO move to commons -->
+  <sql id="operationCondition">
+    <choose>
+      <when test="operation.operator.name().equals('EQUALS')">
+        = #{operation.value}
+      </when>
+      <when test="operation.operator.name().equals('NOT_EQUALS')">
+        != #{operation.value}
+      </when>
+      <when test="operation.operator.name().equals('GREATER_THAN')">
+        &gt; #{operation.value}
+      </when>
+      <when test="operation.operator.name().equals('GREATER_THAN_EQUALS')">
+        &gt;= #{operation.value}
+      </when>
+      <when test="operation.operator.name().equals('LOWER_THAN')">
+        &lt; #{operation.value}
+      </when>
+      <when test="operation.operator.name().equals('LOWER_THAN_EQUALS')">
+        &lt;= #{operation.value}
+      </when>
+      <when test="operation.operator.name().equals('IN')">
+        IN <foreach collection="operation.values" item="value" open="(" separator=", " close=")">#{value}</foreach>
+      </when>
+      <when test="operation.operator.name().equals('LIKE')">
+        LIKE #{operation.value}
+      </when>
+    </choose>
+  </sql>
```

Comment:
@danielkelemen, have you missed writing mapping files? 😆

Reply:
I mean, I kind of like it 😄

Reply:
Well, you only have to write it once, ideally. Unless you have to write specific mapping files for different databases 😆